src/analog.cpp

```diff
@@ -0,0 +1,61 @@
+#include "analog.h"
+
+// assign analog inputs to the correct pins
+AnalogIn acceleratorPedalPin(A6);
+AnalogIn acceleratorSignalPin(A3);
+AnalogIn brakeStatusPin(A1);
+AnalogIn regenerativeBrakingPin(A4);
+
+// initialize the analog inputs' readings
+volatile float acceleratorPedalReading;
+volatile float acceleratorSignalReading;
+volatile float brakeStatusReading;
+volatile float regenerativeBrakingReading;
+
+// read accelerator pedal input
+void readAcceleratorPedal(){
+    acceleratorPedalPin.set_reference_voltage(refrence_voltage);
+    acceleratorPedalReading = acceleratorPedalPin.read_voltage();
+}
+
+// read accelerator signal input
+void readAcceleratorSignal(){
+    acceleratorSignalPin.set_reference_voltage(refrence_voltage);
+    acceleratorSignalReading = acceleratorSignalPin.read_voltage();
+}
+
+// read brake status input
+void readBrakeStatus(){
+    brakeStatusPin.set_reference_voltage(refrence_voltage);
+    brakeStatusReading = brakeStatusPin.read_voltage();
+}
+
+// read regeative braking input
+void readRegeativeBraking(){
+    regenerativeBrakingPin.set_reference_voltage(refrence_voltage);
+    regenerativeBrakingReading = regenerativeBrakingPin.read_voltage();
+}
```

Comment:
Functions look great, however the set_reference_voltage lines could be placed inside of the init function since it is a one time setup thing